File: XcodeExtensionSampleHelper/XcodeExtensionSampleHelper.m
//
//  XcodeExtensionSampleHelper.m
//  XcodeExtensionSampleHelper
//
//  Created by Yoshitaka Seki on 2017/09/09.
//  Copyright © 2017年 takasek. All rights reserved.
//

#import "XcodeExtensionSampleHelper.h"

@implementation XcodeExtensionSampleHelper

// This implements the example protocol. Replace the body of this class with the implementation of this service's protocol.
- (void)upperCaseString:(NSString *)aString withReply:(void (^)(NSString *))reply {
    NSString *response = [aString uppercaseString];
    reply(response);
}

@end

File: XcodeExtensionSampleHelper/XcodeExtensionSampleHelper.swift
//
//  XcodeExtensionSampleHelper.swift
//  XcodeExtensionSampleHelper
//
//  Created by Yoshitaka Seki on 2017/09/09.
//  Copyright © 2017年 takasek. All rights reserved.
//


import Foundation

@objc class XcodeExtensionSampleHelper: NSObject, XcodeExtensionSampleHelperProtocol {
    func write(text: String, to directory: String, reply: @escaping HelperResultHandler) {

        try? text.write(
            toFile: directory + "/outputFile",
            atomically: true, encoding: .utf8
        )

        reply(0)
    }
}

